### ISSUE TITLE
How can I use LZMA-JS for big file?

### ISSUE THREAD

**Issue by zheung** [nmrugg/LZMA-JS#18]
I use LZMA-JS for node.

But I can't find any way to compress big file.
Help me please and thanks!

If I load big file as `Buffer`, it always shows me 'out of memory'.
Have any way to use node's `Stream` on LZMA-JS?

**Comment by jhg** [comment]
The error `out of memory` sound as you need for big file more memory and don't has more memory, I show code of LZMA-c.js and compress method need a str parameter, need all in memory. For your case I think that LZMA-JS need a object for compress in stream, and not load all file in memory only read, for example for each 1024B a call to compress and keep window for next call. And same in decompress. It is possible but not is implemented now.

**Comment by nmrugg** [comment]
Yeah, Node has some serious memory limitations right now. I do plan on 
adding stream support for node, but it's not high on my todo list right 
now. Patches welcome.

This is basically the same as #12, except Node specific.

**Comment by jhg** [comment]
@nmrugg as it is not in your todo list and stream support is important but not critical, I can read code of lzma-c and lzma-d for try patch it (on weekends) and then fix #12 and this

**Comment by nmrugg** [comment]
That would be great, @jhg! Let me know if you have any questions.
